packages/sse/lib/types.tsp

```diff
@@ -0,0 +1,52 @@
+import "@typespec/http/streams";
+
+using Http.Streams;
+
+namespace TypeSpec.SSE;
+
+/**
+ * Describes a stream of server-sent events.
+ *
+ * The content-type is set to `text/event-stream`.
+ *
+ * The server-sent events are described by `Type`.
+ * The event type for any event can be defined by using named union variants.
+ * When a union variant is not named, it is considered a 'message' event.
+ *
+ * @template Type The set of models describing the server-sent events.
+ *
+ * @example Mix of named union variants and terminal event
+ *
+ * ```typespec
+ * model UserConnect {
+ *   username: string;
+ *   time: string;
+ * }
+ *
+ * model UserMessage {
+ *   username: string;
+ *   time: string;
+ *   text: string;
+ * }
+ *
+ * model UserDisconnect {
+ *   username: string;
+ *   time: string;
+ * }
+ *
+ * @TypeSpec.Events.events
+ * union ChannelEvents {
```

Comment:
just curious if we should suggest TypeSpec author to use discriminated union if there's multiple event types? I think client side probably need to handle them differently?